src/Utilities/Workloads.js

```diff
@@ -144,3 +145,48 @@ export const pruneWorkloadsRulesFilters = (localFilters, filterCategories) => {
     return arr;
   }, []);
 };
+
+export const switchSort = (sortIndex, item) => {
+  const rule = item[0]?.rule; // Using optional chaining to check if 'rule' is defined
+  switch (sortIndex) {
+    case 1:
+      return rule?.details || '';
+    case 2:
+      return rule?.total_risk || [];
+    case 3:
+      return rule?.objects?.length || 0;
+    case 4:
+      return rule?.modified || '';
+    default:
+      return 0;
+  }
+};
+
+export const sortWithSwitch = (
+  sortIndex,
+  sortDirection,
+  filteredRows,
+  firstRule
+) => {
+  return sortIndex >= 0 && !firstRule
```

Comment:
nit-pick: can we remove the `firstRule` condition from this function? so that it doesn't complicate the logic here and more adhere to its semantics. instead, you can move this condition to WorkloadsRules component https://github.com/RedHatInsights/ocp-advisor-frontend/pull/653/commits/77277516615d3aba5df120fd2123c6ff890d8f96#diff-ac9d86077d652e9ca2918aa706320d2fff4adf9fb67d2d42752b639a67d3fb18R85: if firstRule is truthy, then don't call sortWithSwitch on the filteredRows.

Reply:
I tried to remove this condition and see how it works without it. Works fine without !firstrule